lib/plugins/eslint.js

```diff
@@ -25,6 +18,13 @@ const babelLoaderUtil = require('../loaders/babel');
  */
 module.exports = function(plugins, webpackConfig) {
     if (webpackConfig.useEslintPlugin) {
+        const forceSync = require('sync-rpc');
+        const hasEslintConfiguration = forceSync(require.resolve('../utils/has-eslint-configuration'));
```

Comment:
Can we move ONLY these 2 lines and put the below 4 back on top? For clarity, I'd like to only move the items that are actually creating the problem.

Reply:
good point - it is likely just the `hasEslintConfiguration` that is the problem since it is USING `sync-rpc`

Reply:
Importing sync-rpc it **is** the main problem for this error. It tried to connect on initialization on import so it is mandatory, to fix this issue.  
Can we only move this one? Maybe, I didnt tried. Maybe you are right, sync-rpc can stay, but I moved all, because I believe, there is no point to init or import all of this things if we never use it. Waste of resources in my opinion. Import and use it only when we need to.

Reply:
> Waste of resources in my opinion. Import and use it only when we need to.

It is a matter of code clarity. We normally expect `require` statements to be at the top of the file. If we start requiring them inline, it'll make the code harder to follow. That's a waste of human resources :). That's why we want to move only the one line needed. That one line actually should have always been moved down here because we are USING a library, not just doing a simple `require`. But the other `require` are standard and should be at the top.

Reply:
@wesoledi the error in the reported issue has a stack trace pointing to line 13, which means the line **calling** `forceSync()`, not the line importing it. So `require('sync-rpc')` is not the issue.

Reply:
Thus, if we migrate to ESM imports in the future (not planned yet, but it might happen at some point as that's the future of the ecosystem), imports are required to be at the top-level. So our existing convention of requiring deps at the top-level will make that migration easier.